framework_tvb/tvb/core/adapters/abcadapter.py

```diff
@@ -523,13 +524,39 @@ def build_adapter(stored_adapter):
             LOGGER.exception(msg)
             raise IntrospectionException(msg)
 
-    def review_operation_inputs(self, parameters):
-        # TODO: implement this for neoforms
+    def review_operation_inputs(self, operation):
         """
         :returns: a list with the inputs from the parameters list that are instances of DataType,\
             and a dictionary with all parameters which are different than the declared defauts
         """
-        return {}, None
+        changed_attr = {}
+        inputs_datatypes = []
+        view_model = self.load_view_model(self, operation)
+        form_model = self.get_view_model_class()()
+        form_fields = self.get_form_class()().fields
+
+        for field in form_fields:
+            if not isinstance(field, TraitDataTypeSelectField):
+                attr_form = getattr(form_model, field.name)
+                attr_vm = getattr(view_model, field.name)
+                if attr_vm != attr_form:
+                    changed_attr[field.label] = attr_vm
+            else:
+                attr_vm = getattr(view_model, field.name)
+                data_type = ABCAdapter.load_entity_by_gid(attr_vm)
+                inputs_datatypes.append(data_type)
+
+        return inputs_datatypes, changed_attr
+
+    def load_view_model(self, adapter_instance, operation):
```

Comment:
This method is duplicated. We also have it in OperationService. 
Have you tested this for simulations? This would probably not work for simulations, since we have to use the SimulatorSerializer there, as in the other method.
Please review this and keep it in a single common place.